packages/utils/src/array-utils/arrayUtils.ts

```diff
@@ -58,6 +58,8 @@ export function reorderArray<T>(array: T[], fromIndex: number, toIndex: number)
 
 export const sortArrayByProperty = <T>(arr: T[], property: string): T[] => {
   return [...arr].sort((a, b) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore -- its ok
     return a[property] - b[property];
   });
 };
```

Comment:
in theory this would make TS happy:

```
export const sortArrayByProperty = <T>(arr: T[], property: keyof T): T[] => {
  return [...arr].sort((a, b) => {
    const valueA = a[property];
    const valueB = b[property];

    if (typeof valueA === 'number' && typeof valueB === 'number') {
      return valueA - valueB;
    } else if (typeof valueA === 'string' && typeof valueB === 'string') {
      return valueA.localeCompare(valueB);
    }

    throw new Error('Property must be a string or number for sorting');
  });
};
```

if we're sure it can only be an integer or a string